apps/qubit/modules/settings/actions/siteInformationAction.class.php

```diff
@@ -27,103 +27,59 @@
-          $this->getUser()->setFlash('notice', $notice);
-
-          $this->redirect('settings/siteInformation');
-        }
-      }
-    }
-
-    $this->populateSiteInformationForm();
-  }
-
-  /**
-   * Populate the site information settings from the database (localized)
-   */
-  protected function populateSiteInformationForm()
-  {
-    // Get site information settings
-    $this->siteTitle = (null !== $siteTitle = QubitSetting::getByName('siteTitle')) ? $siteTitle : new QubitSetting;
-    $this->siteDescription = (null !== $siteDescription = QubitSetting::getByName('siteDescription')) ? $siteDescription : new QubitSetting;
-    $this->siteBaseUrl = (null !== $siteBaseUrl = QubitSetting::getByName('siteBaseUrl')) ? $siteBaseUrl : new QubitSetting;
-
-    // Set defaults values
-    $this->siteInformationForm->setDefaults(array(
-      'site_title' => $this->siteTitle->getValue(array('culture' => $this->culture)),
-      'site_description' => $this->siteDescription->getValue(array('culture' => $this->culture)),
-      'site_base_url' => $this->siteBaseUrl->getValue(array('culture' => $this->culture))
-    ));
-
-    return $this;
+    // Set form decorator
+    $decorator = new QubitWidgetFormSchemaFormatterList($this->form->getWidgetSchema());
+    $this->form->getWidgetSchema()->addFormFormatter('list', $decorator);
+    $this->form->getWidgetSchema()->setFormFormatterName('list');
   }
 
-  /**
-   * Update site information settings (localized)
-   */
-  protected function updateSiteInformationSettings()
+  protected function addField($name)
   {
-    $thisForm = $this->siteInformationForm;
-
-    // Get Site Title
-    $siteTitle = $thisForm->getValue('site_title');
-    $siteTitleSetting = QubitSetting::getByName('siteTitle');
-
-    // Create new QubitSetting if site_title doesn't already exist
-    if (null === $siteTitleSetting)
+    switch ($name)
     {
-      $siteTitleSetting = QubitSetting::createNewSetting('siteTitle', null, array('scope'=>'site_information', 'deleteable'=>false));
-    }
-    $siteTitleSetting->setValue($siteTitle);
-    $siteTitleSetting->save();
+      case 'siteTitle':
+        $this->form->setWidget($name, new sfWidgetFormInput);
+        $this->form->setValidator($name, new sfValidatorString(array('required' => false)));
+        $labelText = $this->i18n->__('Site title');
+        $this->form->getWidgetSchema()->$name->setLabel($labelText);
 
-    // Save Site Description
-    $siteDescription = $thisForm->getValue('site_description');
-    $siteDescSetting = QubitSetting::getByName('siteDescription');
+        break;
 
+      case 'siteDescription':
+        $this->form->setWidget($name, new sfWidgetFormInput);
+        $this->form->setValidator($name, new sfValidatorString(array('required' => false)));
+        $labelText = $this->i18n->__('Site description');
+        $this->form->getWidgetSchema()->$name->setLabel($labelText);
```

Comment:
Nitpick: The `$labelText` declaration isn't necessary as it is only used the one time.